src/lib/Libifl/pbs_loadconf.c

```diff
@@ -933,18 +958,59 @@ __pbs_loadconf(int reload)
 		goto err;
 	}
 
-#ifndef WIN32
 	if ((gvalue = getenv(PBS_CONF_AUTH)) != NULL) {
-		if (!strcasecmp(gvalue, "MUNGE")) {
-			pbs_conf.auth_method = AUTH_MUNGE;
-		} else if (!strcasecmp(gvalue, "GSS")) {
-			pbs_conf.auth_method = AUTH_GSS;
-		} else {
-			fprintf(stderr, "pbsconf error: illegal value for %s\n",PBS_CONF_AUTH);
+		char *value = convert_string_to_lowercase(gvalue);
+		if (value == NULL)
+			goto err;
+		memset(pbs_conf.auth_method, '\0', sizeof(pbs_conf.auth_method));
+		strcpy(pbs_conf.auth_method, value);
+		free(value);
+	}
+	if ((gvalue = getenv(PBS_CONF_ENCRYPT_METHOD)) != NULL) {
+		char *value = convert_string_to_lowercase(gvalue);
+		if (value == NULL)
+			goto err;
+		memset(pbs_conf.encrypt_method, '\0', sizeof(pbs_conf.encrypt_method));
+		strcpy(pbs_conf.encrypt_method, value);
+		free(value);
+	}
+	if ((gvalue = getenv(PBS_CONF_ENCRYPT_MODE)) != NULL) {
+		if (sscanf(gvalue, "%u", &uvalue) == 1)
+			pbs_conf.encrypt_mode = uvalue;
+		if (pbs_conf.encrypt_mode < ENCRYPT_DISABLE || pbs_conf.encrypt_mode > ENCRYPT_ALL) {
+			fprintf(stderr, "pbsconf error: invalid PBS_ENCRYPT_MODE value: %d\n", uvalue);
+			goto err;
+		}
+	}
+	if ((gvalue = getenv(PBS_CONF_SUPPORTED_AUTH_METHODS)) != NULL) {
+		char *value = convert_string_to_lowercase(gvalue);
+		if (value == NULL)
+			goto err;
+		free_string_array(pbs_conf.supported_auth_methods);
+		pbs_conf.supported_auth_methods = break_comma_list(value);
+		if (pbs_conf.supported_auth_methods == NULL) {
+			free(value);
+			goto err;
+		}
+		free(value);
+	}
+
+	if (pbs_conf.supported_auth_methods == NULL && strcmp(pbs_conf.auth_method, AUTH_RESVPORT_NAME) != 0) {
```

Comment:
@hirenvadalia this would imply that the supported_auth_methods is populated at the client side as well. We can reduce another point of data duplication and make it mandatory only for the sever, no?

Reply:
@subhasisb So if I understand your point right then you are saying don't mandate supported_auth_methods in PBS config (or as env) on client-side, and just load given auth method and/or encrypt method if supported_auth_methods is not given, right?
If Yes, then yes we can do that but since PBS config parser code is shared by client and server both, it won't be only for the client, it will also apply to the server also.

Reply:
@subhasisb As per our offline discussion, I have changed code please have a look into latest commit.